qiime2_TagSequencingReport.Rmd

```diff
@@ -0,0 +1,383 @@
+
+```{r, fig.width=6, fig.height=4}
+ggplot(s, aes(x=Read_Counts)) +
+    geom_histogram(aes(fill=SampleType), binwidth=1000) +
+    geom_vline(xintercept = min_reads, color="black", linetype="dashed") +
+    theme_classic() +
+    theme_bw() + 
+    xlab("Number of reads in sample") +
+    ylab("Number of samples")
+```
+
+\newpage
+
+## Whole samples that are above the `r min_reads` read count threshold
+
+```{r}
+### Whole samples discarded that are below the read count threshold of min_reads
+s$Keep <- s$Read_Counts > min_reads
+s$KeepLabel <- factor(ifelse(s$Keep, "Keep", "Discard"))
+keep_table <- table(s[, color_by], s$KeepLabel)
+pander(keep_table)
+```
+
+\newpage
+
+# Taxonomic heatmap
+
+```{r}
+prop_cut <- 0.01
+satu_limit <- 0.4
+```
+
+The heatmap was generated from the most specific taxonomic assignments. Each column represents one sample and each row represents one taxon. Taxa were included in the chart if the abundance in any sample exceeded `r 100*prop_cut`%. 
+
+The chart is colored white if taxa were not observed in the sample, dark blue if taxa were observed at very low abundance. This allows the reader to quickly survey presence/absence. Abundance values exceeding `r 100*satu_limit`% are colored red, indicating an extremely dominant species.
+
+```{r, fig.height=12, fig.width=16}
+s_toPlot <- s[s$Keep,] %>%
+  mutate(study_group=as.character(study_group))
+cts_toPlot <- counts[, s_toPlot$SampleID]  
+a <- simplify_assignments(adf, rank1="Phylum", rank2="Species")
+summed_cts <- rowsum(cts_toPlot, a) 
+summed_props <- sweep(summed_cts, 2, colSums(summed_cts), "/") 
+grps <- c(color_by, shape_by)
+
+### heatmap with a gap
+heatmap_grouped(summed_props, s_toPlot, grps=grps, thre=0.01, option=2, prop_cut = prop_cut, satu_limit=satu_limit)
+```
+
+\newpage
+
+# Alpha Diversity
+
+Alpha diversity was assessd by the number of observed OTUs and Shannon index.
+
+## Number of observed OTUs
+
+```{r, fig.height=3}
+alpha_measure <- "observed_otus"
+
+s %>% filter(Keep) %>%
+  filter(!is.na(time_collected)) %>%
+  ggplot(aes(x=eval(parse(text=color_by)), y=eval(parse(text=alpha_measure)), color=eval(parse(text=color_by)))) +
+  geom_boxplot() +
+  labs(y=alpha_measure, x=color_by, color=color_by) +
+  theme_bw() +
+  theme(axis.text.x=element_text(angle=-25, hjust= .1)) +
+  scale_color_brewer(palette = "Set2")
+```
+
+## Shannon Index
+
+```{r, fig.height=3}
+alpha_measure <- "shannon"
+
+s %>% filter(Keep) %>%
+  filter(!is.na(time_collected)) %>%
+  ggplot(aes(x=eval(parse(text=color_by)), y=eval(parse(text=alpha_measure)), color=eval(parse(text=color_by)))) +
+  geom_boxplot() +
+  labs(y=alpha_measure, x=color_by, color=color_by) +
+  theme_bw() +
+  theme(axis.text.x=element_text(angle=-25, hjust= .1)) +
+  scale_color_brewer(palette = "Set2")
+```
+
+\newpage 
+
+# Beta diversity
+
+Similarity between samples was assessed by unweighted and weighted UniFrac distances. 
+
+## Unweighted UniFrac distances
+
+```{r}
```

Comment:
This section can be hardcoded in. Unless maybe you would like to factor this out into a function so it's not repeated in line 342?

Reply:
I think it would be good if I could run a for loop in .Rmd so that I can handle repeated alpha measures and beta measures more nicely.